pkg/webservice/handlers.go

```diff
@@ -746,6 +752,11 @@ func getQueueApplications(w http.ResponseWriter, r *http.Request) {
 		appsDao = append(appsDao, getApplicationDAO(app))
 	}
 
+	if checkHeader(r.Header, "Content-Encoding", "gzip") {
```

Comment:
This PR aims to enhance only `getQueueApplication`. However, I'd like to open the room to discuss whether we should bring this enhancement to all restful APIs. @targetoee WDYT?

Reply:
That's a good idea. It can be an optional choice for user to provide more flexibility.
Some APIs typically don't return large amounts of data in common cases, so it may be necessary to discuss which ones require this functionality.

Reply:
> it may be necessary to discuss which ones require this functionality.

The improvement you propose is a kind of infra to our Restful APIs, so I prefer to bring such benefit to all APIs if there is no obvious side-effect or cost.

Reply:
That is exactly why I would make the two functions so we can easily apply it to any REST end points. Anything that sends more than a single IP packet as the response can benefit. 
For the streaming API, which uses really small messages that fit in a single IP packet, compressing might be more overhead than the gains we get so that one might not be a candidate everything else is.